.github/workflows/master_dev_checks.yml

```diff
@@ -0,0 +1,409 @@
+---
+  name: Best Practices Check
+  on:
+    workflow_call:
+    
+  jobs:
+
+    dev-checks:
+      runs-on: non-build
+      steps:
+        - name: Dump GitHub context
+          env:
+            GITHUB_CONTEXT: ${{ toJson(github) }}
+          run: |
+            echo "$GITHUB_CONTEXT"
+
+        - uses: actions/checkout@v3
+          with:
+            token: ${{ secrets.CIPIPELINE_NEXUS_PASSWORD  }}
+            fetch-depth: 0
+
+        - name: Get branch name (merge)
+          if: github.event_name != 'pull_request'
+          shell: bash
+          run: |
+            echo "SOURCE_BRANCH=$(echo ${GITHUB_REF#refs/heads/} | tr / -)" >> $GITHUB_ENV
+
+        - name: Get branch name (pull request)
+          if: github.event_name == 'pull_request'
+          shell: bash
+          run: |
+            echo "SOURCE_BRANCH=$(echo ${{ github.event.pull_request.head.ref }} | tr / -)" >> $GITHUB_ENV
+            echo "TARGET_BRANCH=$(echo ${GITHUB_BASE_REF} | tr / -)" >> $GITHUB_ENV
+            echo "GITHUB_TOKEN=${{ secrets.CIPIPELINE_NEXUS_PASSWORD  }}" >> $GITHUB_ENV
+        
+        - name: Check Format
+          env:
+            PR_TITLE: ${{ github.event.pull_request.title }}
+          run: |
+            if echo "${{ env.SOURCE_BRANCH }}" | grep -Eq '^revert-'; then
+                echo "Revert branch, skipping Format check"
+                exit 0
+            fi
+            if echo "${{ env.SOURCE_BRANCH }}" | grep -Eq '^(TS-|SRE-|TDEVOPS-|TOPS-)'; then
+              echo "Branch name starts with 'TS-' or 'SRE-' or 'TDEVOPS-' or 'TOPS-'"
```

Comment:
If the variable works. then lets just print the variable value in message instead of hard-coding the project names.